src/components/GameStats.tsx

```diff
@@ -33,6 +33,19 @@ export const GameStats = (props: { noSpace?: boolean }) => {
     [phaseDeadline]
   );
 
+  const endGameTimestamp = useMemo(
+    () =>
+      phase === GamePhase.GUESS && phaseDeadline?.toNumber()
```

Comment:
Please change useLupiContract to return both deadlines and use those values instead of assuming the time.

Changing this code:

```  const phaseDeadline = useMemo(() => {
    if (contractState) {
      const { blockTimestamp, guessDeadline, revealDeadline } = contractState;
      if (blockTimestamp.lte(guessDeadline)) {
        return guessDeadline;
      } else if (blockTimestamp.lte(revealDeadline)) {
        return revealDeadline;
      } else {
        return BigNumber.from(0);
      }
    } else {
      return BigNumber.from(0);
    }
  }, [contractState]);```